src/Microsoft.AspNetCore.SignalR.Redis/Internal/MsgPackUtil.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using MessagePack;
+
+namespace Microsoft.AspNetCore.SignalR.Redis.Internal
+{
+    internal static class MsgPackUtil
+    {
+        public static int ReadArrayHeader(ref ReadOnlyMemory<byte> data)
+        {
+            var arr = GetArray(in data);
+            var val = MessagePackBinary.ReadArrayHeader(arr.Array, arr.Offset, out var readSize);
+            data = data.Slice(readSize);
+            return val;
+        }
+
+        public static int ReadMapHeader(ref ReadOnlyMemory<byte> data)
+        {
+            var arr = GetArray(in data);
+            var val = MessagePackBinary.ReadMapHeader(arr.Array, arr.Offset, out var readSize);
+            data = data.Slice(readSize);
+            return val;
+        }
+
+        public static string ReadString(ref ReadOnlyMemory<byte> data)
+        {
+            var arr = GetArray(in data);
+            var val = MessagePackBinary.ReadString(arr.Array, arr.Offset, out var readSize);
+            data = data.Slice(readSize);
+            return val;
+        }
+
+        public static byte[] ReadBytes(ref ReadOnlyMemory<byte> data)
+        {
+            var arr = GetArray(in data);
```

Comment:
is this when you have to pass in explicitly?

Reply:
Apparently. I was getting a compiler error without it. I'll try removing it and verify it.